### DISCUSSION TITLE
WYSIWYG - Insert Code Block problem

### DISCUSSION THREAD

**Discussion by ulasyuksel1979** [requarks/wiki#2971] [Help / Questions] [answered]
Hi,
I've a problem with WYSIWYG editor. I'm trying to use "Insert Code Block" section but the script content can not be rendered.
This is what I've coded with Java-script
![image](https://user-images.githubusercontent.com/59475285/105971087-2a408180-609b-11eb-8021-43bda0ded8a2.png)
This is the resulting rendered output.
![image](https://user-images.githubusercontent.com/59475285/105971227-59ef8980-609b-11eb-940c-ab7623eebe3b.png)

What is wrong with my usage?

Version:2.5.170
Docker Container (Linux)
Node.js 14.14.0
PostgreSQL 13.1

**Comment by adjurk** [comment]
Are you trying to display a JavaScript alert? JS code can be included in *Page -> Scripts* within `<script>` tags.

![image](https://user-images.githubusercontent.com/13603158/106132478-49efac80-6164-11eb-99be-ebced6f4e81e.png)

Note that this requires at least [version 2.5.117](https://docs.requarks.io/releases#major-features-25117).

**Reply by ulasyuksel1979** [reply]
I'm sorry, I've misleaded everyone mentioning javascript code. 
Basically, I want to put an Iframe inside a page created with WYSIWYG editor.  Such as, ""<iframe src="https://XXXXX/YYYY" height="1500" width="1000" style="border:none;"></iframe>"" 
When using Markdown editor, it is possible to add and "HTML Iframe" compenent. However, WYSIWYG editor, I could not placed an Iframe windows inside the page.

**Reply by adjurk** [reply]
I see. In that case, try enabling iframes in *Rendering -> Security -> html to html -> Allow iframes*:

![image](https://user-images.githubusercontent.com/13603158/106268325-3c512a00-622b-11eb-8c6c-1915d3b74dc1.png)

**Reply by ulasyuksel1979** [reply]
I've made that configuration and it seems that it is only working for Markdown editor.

**Comment by ulasyuksel1979** [comment]
Are there any update regarding with our requirement. I've tried lots of thing but I could not accomplished:(

**Comment by NGPixel** [comment, marked as answer]
You can't insert arbitrary HTML code in the Visual Editor. Use the Markdown editor instead for this purpose.
